### ISSUE TITLE
Investigate whether format verification is working

### ISSUE THREAD

**Issue by tmorrell** [inveniosoftware/datacite#76]
Package version (if known): v1.1.2

## Describe the bug
The date format verification https://github.com/inveniosoftware/datacite/blob/b15db91e1231135e5f2dba0cadca0c72ede037cf/datacite/schemas/datacite-v4.3.json#L80 may not be used by default https://python-jsonschema.readthedocs.io/en/latest/validate/#validating-formats or be implemented correctly.

**Comment by edager** [comment]
Hi @tmorrell,  

Although it's allowed to use [format for notational purposes](https://json-schema.org/understanding-json-schema/reference/string#format). However if it should actually validate the format, a validator needs to be supplied while validating. As far as I can tell, this is not happening, which in principle means that e.g. `"publicationYear"` is only being validated as being a string (see dummy example below). It might be useful to have a regex instead of these formats that aren't properly validated?         

```python
from jsonschema import validate

schema = {
    "additionalProperties": False,
    "definitions": {
        "date": {
            "type": "string",
            "anyOf": [
                {"format": "year"}
            ]
        }
    },

    "type": "object",
    "properties": {
        "publicationYear": {
            "type": "string",
            "format": "year"
        }
    },
    "required": [
        "publicationYear"
    ]
}

tests = [
    {"publicationYear":"1234"},
    {"publicationYear":"year"},
    {"publicationYear":""},
]
for test in tests:
    if validate(instance=test, schema=schema) is None:
        print(f"{test['publicationYear']} is a valid year format")
```

**Comment by tmorrell** [comment]
This is fixed in the 4.5 schema
